### ISSUE TITLE
[clang-tidy] False positive bugprone-non-zero-enum-to-bool-conversion with custom operator

### ISSUE THREAD

**Issue by chrchr-github** [llvm/llvm-project#65315]
~~~c++
enum E {
    E0 = 0x1,
    E1 = 0x2,
    E2 = 0x4
};

E operator&(E a, E b) { return static_cast<E>(a & b); }

void f(E e) {
    if (e & E1) {}
}
~~~
~~~
<[source>:10:9: warning: conversion of 'E' into 'bool' will always return 'true', enum doesn't have a zero-value enumerator [bugprone-non-zero-enum-to-bool-conversion]
   10 |     if (e & E1) {}
      |         ^
[<source>:1:6: note: enum is defined here]
    1 | enum E {
      |      ^
~~~
https://godbolt.org/z/nv3PxYdM7

**Comment by PiotrZSL** [comment]
From check documentation:
```
May produce false positives if the enum is used to store other values (used as a bit-mask or zero-initialized on purpose). To deal with them, // NOLINT or casting first to the underlying type before casting to bool can be used.
```

But I agree that this specific case probably could be excluded on check level.
Expect this to be fixed in Clang-tidy 18 timeframe.

**Comment by HerrCai0907** [comment]
Do you think it will be better instead of treating this case as fake positive?

```c++

using E_BitMap = uint32_t;
enum E {
    E0 = 0x1,
    E1 = 0x2,
    E2 = 0x4
};

E_BitMap operator&(E a, E b) { return static_cast<E_BitMap>(a & b); }

void f(E e) {
    if (e & E1) {}
}

```
